{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name}}/validation/notebooks/ModelValidation.py

```diff
@@ -172,15 +172,38 @@ def get_targets_from_recipe():
 
 # COMMAND ----------
 
-eval_result = None
-err = None
+# helper methods
+def get_run_link(run_info):
+    return "[Run](#mlflow/experiments/{0}/runs/{1})".format(
+        run_info.experiment_id, run_info.run_id
+    )
 
 
-def log_to_model_description(run, success):
-    run_info = run.info
-    run_link = "[Run](#mlflow/experiments/{0}/runs/{1})".format(
-        run_info.experiment_id, run_info.run_id
+def get_training_run(model_name, model_version):
+    model_versions = client.search_model_versions(f"name='{model_name}'")
+    training_run_id_optional = [
+        version.run_id for version in model_versions if version.version == model_version
+    ]
+    if not training_run_id_optional:
+        print("Unable to find the training run!")
+        return None
+    return mlflow.get_run(run_id=training_run_id_optional[0])
+
+
+def generate_run_name(training_run):
+    return None if not training_run else training_run.info.run_name + "-validation"
```

Comment:
Maybe we shouldn't assume there only one validation run for a training run. What happens if users want to try out different metrics or different validation sets, but don't want to go through retraining? Or if the validation run fails with a retriable error (for whatever reason)?

Reply:
Discussed offline - this would work fine even in presence of multiple validation runs. LGTM